### ISSUE TITLE
ENH: Add "duration stamp" parser similar to `strptime`

### ISSUE THREAD

**Issue by corneliusroemer** [pola-rs/polars#7397]
### Problem description

A format not as common as time stamps but still common enough to potentially warrant a parser are "duration stamps": e.g. what stop watches show.

Examples that currently cannot be parsed with `strptime`:
- `80:00` meaning 80 minutes
- `-04:00` minus 4 hours
- `104:00:00` 104 hours

Maybe such a parser already exists? I'm not aware of one. It would benefit from standard solution as edge cases can be hairy (there may be at least 2 digits for hours, but if it's say 104 hours then hours can be 3 digits; minus signs; wrapping; whether to accept multi-wrapping, e.g. `23:483` etc.

In any case, if someone has a good idea, contribute to the time stamp usage guide, see PR: https://github.com/pola-rs/polars-book/pull/275

A StackOverflow question inspired this: https://stackoverflow.com/a/75664067/7483211

The accepted answer fails things like `80:00.000` (first solution) and `104:00.000` (first and second).

**Comment by aldanor** [comment]
Agreed, especially because functions like `groupby_rolling` parse those strings to durations internally and it looks like you have no way to use the same parser: `1ns`, `5m`, etc (you can parse them via `pd.to_timedelta()` as it stands but that's just a workaround).

**Comment by corneliusroemer** [comment]
@aldanor Can you show the `pd.to_timedelta()` in action? I'm curious what that would look like.

**Comment by stinodego** [comment]
To clarify:

We use the `chrono` crate for parsing dates, times, and datetimes. This crate does not offer functionality for parsing duration strings. We have implemented a rudimentary parser that can parse strings like `"1d20h"`, but this is not a fully featured parser.

We need to add full parsing functionality to the Rust side, similar to how chrono parses datetimes.

This functionality then needs to be exposed in two ways (both for Rust and Python):
* A `to_duration` method on the DateTime namespace
* Support for `dtype=Duration` for the `strptime` method in the DateTime namespace

Help is very much appreciated on this issue!

**Comment by filabrazilska** [comment]
Hi,
I've taken a shot at this in https://github.com/pola-rs/polars/compare/main...filabrazilska:polars:to_duration
I'm allowing up to three digits for hours, minutes and seconds when converting a string to a Duration object. I needed to cut the parser out of the chrono crate because the `Parsed` object there checks the number of hours, seconds, minutes fits their respective range (24 or 60).
Does it make sense to finish the commit (add tests, docs, etc.) for pulling or is this functionality so niche it doesn't afford the ~250lines that would need to be added?

**Comment by stinodego** [comment]
Hi @filabrazilska , best thing to do is to just open a PR - then we can take a look at it and discuss.

**Comment by terryluan12** [comment]
Hey there! Is this issue still open? If so, I'd love to give it a shot. Please let me know

**Comment by ayushshah37** [comment]
Hey @terryluan12 , are you still working on this ?  I'd like to give it a shot

**Comment by Crypto-Spartan** [comment]
@ayushshah37 I have absolutely zero authority here, but given that there hasn't been any PRs for this in the 7 months since their message, I'd say you're probably good to give it a shot.
